tests/integration/test_random_creation.py

```diff
@@ -33,6 +35,366 @@ def test_randn():
+            # without dtype:
+            # np.random.randint(5, 10000, None).__class__
+            #     <class 'int'>
+            # num.random.randint(5, 10000, None).__class__
+            #     <class 'cunumeric.array.ndarray'>
+        ),
+    ),
+    0,
+    (20, 50, 4),
+]
+
+
+@pytest.mark.parametrize("low, high", LOW_HIGH, ids=str)
+@pytest.mark.parametrize("size", SIZES, ids=str)
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        np.int64,
+        pytest.param(
+            np.uint16,
+            marks=pytest.mark.xfail(
+                reason="cuNumeric raises NotImplementedError"
+            ),
+        ),
+    ],
+    ids=str,
+)
+def test_randint(low, high, size, dtype):
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low=low, high=high, size=size, dtype=dtype
+    )
+    assert arr_np.shape == arr_num.shape
+    # skip distribution assert for small arrays
+    if arr_np.size > 1024:
+        assert_distribution(
+            arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+        )
+
+
+@pytest.mark.xfail(reason="cuNumeric raises NotImplementedError")
+@pytest.mark.parametrize("size", (1024, 1025))
+def test_randint_bool(size):
+    """Test randint with boolean output dtype."""
+    arr_np, arr_num = gen_random_from_both("randint", 2, size=size, dtype=bool)
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+    # NumPy pass
+    # cuNumeric LEGATE_TEST=1 or size > 1024:
+    # NotImplementedError: type for random.integers has to be int64 or int32
+    # or int16
+
+
+@pytest.mark.parametrize("low, high", LOW_HIGH, ids=str)
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_random_integers(low, high, size):
+    arr_np, arr_num = gen_random_from_both(
+        "random_integers", low=low, high=high, size=size
+    )
+    assert arr_np.shape == arr_num.shape
+    # skip distribution assert for small arrays
+    if arr_np.size > 1024:
+        assert_distribution(
+            arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+        )
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_random_sample(size):
+    arr_np, arr_num = gen_random_from_both("random_sample", size=size)
+    assert arr_np.shape == arr_num.shape
+    # skip distribution assert for small arrays
+    if arr_np.size > 1024:
+        assert_distribution(
+            arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+        )
+
+
+class TestRandomErrors:
+    def assert_exc_from_both(self, func, context, *args, **kwargs):
+        with context:
+            getattr(np.random, func)(*args, **kwargs)
+        with context:
+            getattr(num.random, func)(*args, **kwargs)
+
+    @pytest.mark.parametrize(
+        "seed, expected_exc",
+        [
+            (-100, pytest.raises(ValueError)),
+            (12.0, pytest.raises(TypeError)),
+            ("abc", pytest.raises(TypeError)),
```

Comment:
It's unusual to pass the value of `pytest.raises` around, I'd suggest keeping the configuration here simpler and more declarative, i.e. just the execption type, and move the `pytest.raises` into `assert_exc_from_both`

Reply:
It's not unusual though, and AFAICT this is the only way to do it: https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising

Reply:
That's an interesting usage I had not encountered before, but I also think that that is for a different situation. That technique is useful when you have a situation where you need to handle *no exception* cases together with *some exception* cases, in the same test. But that's not the situation here, AFAICT? All the cases in this test alwasy result in *some exception*, so the parameterization can just pass the exception itself.

Reply:
Specifically, I am suggesting:
```
    def assert_exc_from_both(self, func, exc, *args, **kwargs):
        with pytest.rasises(exc):
            getattr(np.random, func)(*args, **kwargs)
        with pytest.raises(exc):
            getattr(num.random, func)(*args, **kwargs)
```
and
```
    @pytest.mark.parametrize(
        "seed, expected_exc",
        [
            (-100, ValueError),
            (12.0, TypeError),
            ("abc", TypeError),
        ],
        ids=lambda x: f" {str(getattr(x, 'expected_exception', x))} ",
    )
```
If you think that in the future it is very likely that this test will also need to handle some case with ***no*** exception, then that would be a reason to leave it as it is. But otherwise I think the version here will be more familiar and approachable to future readers of the code